File: quick_sort.cpp
#include <iostream>
using namespace std;
void quickSort(int a[], int left, int right) {
    int i = left;
    int j = right;
    int pivot = a[(left + right) / 2];
    while (i <= j) {
        while (a[i] < pivot) {
            i++;
        }

        while (a[j] > pivot) {
            j--;
        }

        if (i <= j) {
            swap(a[i], a[j]);
            i++;
            j--;
        }
    }
    if (left < j) {
        quickSort(a, left, j);
    }

    if (i < right) {
        quickSort(a, i, right);
    }
}
int main() {
    int n;
    cin >> n;
    int a[100];
    for (int i = 0; i < n; i++) {
        cin >> a[i]; }
    quickSort(a, 0, n);
    for (int i = 0; i < n; i++) {
        cout << a[i] << " ";}
    return 0;
}

File: quick_sort.py
def quick_sort(a,left,right):
    pivot = a[(left+ right ) // 2]
    i = left 
    j = right
    while i <=j :
        while a[i] <pivot:
            i+=1
        while a[j]> pivot:
            j-=1
        if i<=j:
            tam = a[i]
            a[i] = a[j]
            a[j] = tam
            i+=1
            j-=1
    if left < j :
        quick_sort(a, left , j)
    if i < right :
        quick_sort(a, i, right)            
a=[]
n=int(input("nhap n: "))
for i in range(0,n,1):
    x = input()
    a.append(x)
quick_sort(a,0,n-1)
print(a)  
     